cmd/kubeadm/app/cmd/phases/reset/removeusers.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
+	"k8s.io/kubernetes/cmd/kubeadm/app/util/users"
+)
+
+// NewRemoveUsersPhase creates a kubeadm workflow phase for remove-etcd-member
+func NewRemoveUsersPhase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "remove-users",
+		Short:        "Remove users and groups created for control-plane.",
+		Long:         "Remove users and groups created for control-plane.",
+		Run:          runRemoveUsersPhase,
+		InheritFlags: []string{},
+	}
+}
+
+func runRemoveUsersPhase(c workflow.RunData) error {
+	if err := users.RemoveUsersAndGroups(); err != nil {
+		return errors.Wrap(err, "failed to remove control-plane users")
+	}
+
+	return nil
+}
```

Comment:
Lets not add a new phase for this yet. Maybe for the beta.
For cp nodes when the ClusterConfiguration is downloaded from the cluster and if the fg is set the operation can run.

Reply:
instead of adding a new phase should this be part of https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go and do we want to do that in alpha?

Reply:
I see your comment bellow that this should always run?
It seems better for it to only happen if the fg is on, because we do not want to touch the system files. 

Also if you think a phase is a better location organization wise we should hide it for now. See the Hidden flag when creating a Phase object.

Reply:
I don't think phase is better location organization wise, this probably is better in node cleanup. Ill move it there.

Reply:
Ok updated to:
1. remove the phase
2. respect the featureGate flag